apps/web/lib/team/[slug]/[type]/getServerSideProps.tsx

```diff
@@ -68,47 +80,43 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     booking = await getBookingForReschedule(`${rescheduleUid}`, session?.user?.id);
   }
 
-  const org = isValidOrgDomain ? currentOrgDomain : null;
-  // We use this to both prefetch the query on the server,
-  // as well as to check if the event exist, so we c an show a 404 otherwise.
-  const eventData = await ssr.viewer.public.event.fetch({
-    username: teamSlug,
-    eventSlug: meetingSlug,
-    isTeamEvent: true,
-    org,
-    fromRedirectOfNonOrgLink: context.query.orgRedirection === "true",
-  });
-
-  if (!eventData) {
-    return {
-      notFound: true,
-    } as const;
-  }
+  const ssr = await ssrInit(context);
+  const fromRedirectOfNonOrgLink = context.query.orgRedirection === "true";
+  const isUnpublished = team.parent ? !team.parent.slug : !team.slug;
 
   return {
     props: {
       eventData: {
-        entity: eventData.entity,
-        length: eventData.length,
-        metadata: eventData.metadata,
+        entity: {
+          fromRedirectOfNonOrgLink,
+          considerUnpublished: isUnpublished && !fromRedirectOfNonOrgLink,
+          orgSlug: team.parent?.slug,
+          teamSlug: team.slug,
+        },
+        length: team.eventTypes[0].length,
```

Comment:
We can just use the first one?

Reply:
Nevermind. I see the query pulling back event types by slug and so we should only get 1.